messagesView.js

```diff
@@ -0,0 +1,151 @@
+class MessagesView {
+  constructor(model, api, user) {
+    this.model = model;
+    this.api = api;
+    this.user = user;
+
+    this.mainContainerEl = document.querySelector('#main-container');
+
+    this.addMessageInput = document.querySelector('#add-message-input');
+    this.addMessageButton = document.querySelector('#add-message-button');
+
+    this.addMessageButton.addEventListener('click', () => {
+      this.addMessage(this.addMessageInput.value);
+      this.addMessageInput.value = "";
+    });
+
+    this.signUpButton = document.querySelector('#sign-up-button');
+
+    this.signUpButton.addEventListener('click', () => {
+      document.querySelector('#sign-up-container').innerHTML = "";
+      this.displaySignUpForm();
+    });
+
+    this.loginButton = document.querySelector('#login-button');
+
+    this.loginButton.addEventListener('click', () => {
+      document.querySelector('#login-container').innerHTML = "";
+      this.displayLoginForm();
+    });
+  }
+
+  addMessage(text) {
+    const userId = this.user.getUserId();
+    const sessionKey = this.user.getSessionKey();
+
+    this.api.postMessage(userId, sessionKey, text, (data) => {
+      console.log(data);
+      this.displayMessagesFromApi();
+    });
+  }
+
+  displayTime(timeString) {
+    let date = timeString.substring(0, 10);
+    let time = timeString.substring(11, 19);
+    return time + "  " + date;
+  }
+
```

Comment:
I like that you added so many elements to your messages. I didn't get round to doing this.